src/osk_gui.cpp

```diff
@@ -244,9 +244,9 @@ static std::unique_ptr<NWidgetBase> MakeTopKeys()
 	auto hor = std::make_unique<NWidgetHorizontal>();
 	hor->SetPIP(0, INTER_KEY_SPACE, 0);
 
-	AddKey(hor, TOP_KEY_PADDING, 6 * 2, WWT_TEXTBTN,    WID_OSK_CANCEL,    STR_BUTTON_CANCEL);
-	AddKey(hor, TOP_KEY_PADDING, 6 * 2, WWT_TEXTBTN,    WID_OSK_OK,        STR_BUTTON_OK    );
-	AddKey(hor, TOP_KEY_PADDING, 2 * 2, WWT_PUSHIMGBTN, WID_OSK_BACKSPACE, SPR_OSK_BACKSPACE);
+	AddKey(hor, TOP_KEY_PADDING, 6 * 2, WWT_TEXTBTN,    WID_OSK_CANCEL,    WidgetData{.string = STR_BUTTON_CANCEL});
+	AddKey(hor, TOP_KEY_PADDING, 6 * 2, WWT_TEXTBTN,    WID_OSK_OK,        WidgetData{.string = STR_BUTTON_OK});
+	AddKey(hor, TOP_KEY_PADDING, 2 * 2, WWT_PUSHIMGBTN, WID_OSK_BACKSPACE, WidgetData{.sprite = SPR_OSK_BACKSPACE});
```

Comment:
Would it make sense for `AddKey()` to return a reference to the widget it just created, so that you can then do `AddKey(...).SetString(...)`?

Reply:
I don't think it makes much sense. For most keys we need to construct the `WidgetData` anyhow due to the `NWidgetLeaf` constructor, so instead of writing the memory once we need to do it twice. Next to that we `std::move` the `NWidgetLeaf`'s `unique_ptr`, so are we still in control and should we conceptually return a reference to the object we released?

On the other hand, doing it that way would make it possible to split the last commit up into smaller commits. It might also look prettier.

I dislike being torn between conceptually less appealing but visually more appealing.

Reply:
Also, in half of the code paths spacer would be created which does not have `NWidgetCore` as ancestor, so I'd need to return a pointer to the `NWidgetLeaf` (the other code path) and `nullptr`. As such it'd become `AddKey(...)->SetString(...)` or _boom_ when you try to set the text on a spacer.

Reply:
Ah, not so clear, never mind!